Fix UTC Serialization in PDFDate.serialize() Method

Shenile opened the pull request:
<!--
Thanks for your interest in the project. Bugs filed and PRs submitted are appreciated!

Some guidelines are provided like this, in HTML comments, to expedite the code review before merging your contribution.

First, please make sure that you have read the documentation page on fpdf2 development:
https://py-pdf.github.io/fpdf2/Development.html

If you're new to contributing to open source projects,
you might find this free video course helpful: http://kcd.im/pull-request
-->

<!-- What changes are being made? (What feature/bug is being fixed here?) -->
This pull request fixes a bug in the `PDFDate.serialize()` method related to the serialization of UTC dates in PDF metadata. Previously, UTC dates were incorrectly serialized to include timezone information, resulting in formats like "Z00'00'" or "Z15:22". The expected format for UTC dates should simply be "Z".

e.g. Fixes #1261 <!-- This will automatically close issue #1261 once the PR is merged: https://help.github.com/en/articles/closing-issues-using-keywords -->

<!-- Have you done all of these things?  -->
**Checklist**:

- [ ] A unit test is covering the code added/modified by this PR
- [x] This PR is ready to be merged <!-- In your opinion, can this be merged as soon as it's reviewed? Else, this can be turned into a Draft PR -->
- [ ] In case of a new feature, docstrings have been added, with also some documentation in the `docs/` folder
- [x] A mention of the change is present in `CHANGELOG.md`

<!-- Feel free to add additional comments, and to ask questions if some of those guidelines are unclear to you! -->

Lucas-C commented:
Thank you for your contribution @Shenile 👍

There a few minor details preventing this PR to be merged for now:
* `test/linearization.pdf` should not be added as part of this PR. It is not needed, and currently makes the GitHub Actions pipeline fail.
* the reference PDF file `test/image/image_types/image_types_insert_jpg_flatedecode_windows.pdf` has significantly grown, which is a bit strange... Do you have all `fpdf2` dependencies up-to-date ? Maybe try to run `pip install --update . -r test/requirements.txt`

Shenile commented:
> Thank you for your contribution @Shenile 👍
> 
> There a few minor details preventing this PR to be merged for now:
> 
> * `test/linearization.pdf` should not be added as part of this PR. It is not needed, and currently makes the GitHub Actions pipeline fail.
> * the reference PDF file `test/image/image_types/image_types_insert_jpg_flatedecode_windows.pdf` has significantly grown, which is a bit strange... Do you have all `fpdf2` dependencies up-to-date ? Maybe try to run `pip install --update . -r test/requirements.txt`

Thank you for your feedback!

I have removed the `test/linearization.pdf ` file from the PR, as it was not needed. Additionally, I have updated the dependencies by running `pip install --update . -r test/requirements.txt` .

Please let me know if there's anything else I need to address!

Lucas-C commented:
A few unit tests are still failing in the GitHub Actions pipeline:
```
test/image/image_types/test_insert_images.py::test_insert_jpg_jpxdecode FAILED [ 37%]
test/image/image_types/test_insert_images.py::test_insert_jpg_flatedecode FAILED [ 37%]
test/image/image_types/test_insert_images.py::test_insert_png_alpha_dctdecode FAILED [ 38%]
test/layout/test_page_background.py::test_page_background FAILED         [ 43%]
```
Each time it's due to the `/CreationDate`

Could you check if those 4 unit tests pass on the your own development environment @Shenile?

Lucas-C commented:
Hi @Shenile 🙂

Just to get a quick update: do you still want to implement this bugfix?

Lucas-C commented:
Hi @Shenile 

Given your absence of response, I'm planning to finish the work you initiated in this PR soon.
If you still want to complete this, just drop a comment 🙂
